doc/source/scripts/base.py

```diff
@@ -18,87 +70,73 @@ class TableWriter(object):
     Filename relative to source.
     """
 
-    filename = ""
-    include_table = False
-    headings = []
-    preprocess = []
-    postprocess = []
-    sort = True
+    def __init__(
+        self,
+        filename: str = "",
+        include_table: Optional[str] = None,
+        sort: bool = True,
+        input_items: Iterable[Any] = set(),
```

Comment:
```suggestion
        input_items: Iterable[Any] = tuple(),
```

If this can be any iterable, an immutable one is usually safer in case it gets modified.

Reply:
Agreed!